### DISCUSSION TITLE
Mejorar experiencia de usuario para programar en el admin de Cotalker

### DISCUSSION THREAD

**Discussion by ValeMP** [Cotalker/documentation#823] [3. Ideas & Feature Requests]
Hola! 

Queria proponer mejorar la experiencia de usuario para las secciones del admin de Cotalker para programar. 
Hasta el momento solo es una vista de texto simple donde uno debe poner su código, por tanto, sería una mejora para hacer las pruebas y programar directamente (un código simple).

Además, muchas de las veces que hay error en esa sección es por el uso de comillas. No queda muy claro cuales son las comillas que dan errores, por tanto, hay que ir segmentando el código para ir probando y evaluar qué es lo que te da error. Asimismo, los automation log no especifican cuál es el error. Por tanto, sucede varias veces que funciona el código en un archivo .js pero no cuando se pone en Cotalker. 

También propongo que la vista de question exec también sea mejorada, dada que tampoco se puede observar bien la funciones. El espacio para programar es muy pequeño y también se observa  como un texto simple. 

Cuales son las principales razones de porque seria una buena idea:
- menor frustración de usuarios nuevos
- menor tiempo dedicado cuando quiero agregar código
- mayor facilidad de observar errores

Incluso sería genial agregar en la documentación las cosas que no podría sostener el ccjs. Entiendo que una de las razones es porque ese código es almacenado como un String en la base de datos.

Asimismo, sería genial que dentro de un mismo ccjs se pueda hacer uso del pb network request. Así podríamos crear un código más sofisticado en Cotalker y además, podríamos evitar el uso de excesivas etapas. Lo propongo para usarlo como complemento a las etapas.

c ya

**Comment by acedward** [comment]
> Hasta el momento solo es una vista de texto simple donde uno debe poner su código, por tanto, sería una mejora para hacer las pruebas y programar directamente (un código simple).

¿Para campos de COTLang / Question-Exec? 
Hay otros campos?

> Además, muchas de las veces que hay error en esa sección es por el uso de comillas. No queda muy claro cuales son las comillas que dan errores, por tanto, hay que ir segmentando el código para ir probando y evaluar qué es lo que te da error. 

¿Tienes algún ejemplo?

> Asimismo, los automation log no especifican cuál es el error. Por tanto, sucede varias veces que funciona el código en un archivo .js pero no cuando se pone en Cotalker.

Absolutamente. Tenemos mucho que hacer a nivel de logs globales, como también en cada bot que reporte mejor sus fallas. Esta tarea es un poco más difícil, porque significa ir a cada bot y ver cuales pueden ser sus errores, capturarlas y describirlas.

> También propongo que la vista de question exec también sea mejorada, dada que tampoco se puede observar bien la funciones. El espacio para programar es muy pequeño y también se observa como un texto simple.

Todos odiamos esto : )

> Incluso sería genial agregar en la documentación las cosas que no podría sostener el ccjs. 

¿Tienes una idea de cómo podría verse esto?

> Entiendo que una de las razones es porque ese código es almacenado como un String en la base de datos.

Todo se guarda como string! Pero los archivos de código fuente también.
Podemos hacer que se vean mejor

> Asimismo, sería genial que dentro de un mismo ccjs se pueda hacer uso del pb network request. Así podríamos crear un código más sofisticado en Cotalker y además, podríamos evitar el uso de excesivas etapas. Lo propongo para usarlo como complemento a las etapas.

Esta tarea está súper interesante. Nos gustaría, pero requerimos un genio de la programación que nos ayude con esta tarea. Creo que es posible :-)


Gracias por el tremendo feedback.
Vamos a ir creando tareas de algunos de estos issues.

**Reply by ValeMP** [reply]
> ¿Para campos de COTLang / Question-Exec? Hay otros campos?

Me parece que no 

> Además, muchas de las veces que hay error en esa sección es por el uso de comillas. No queda muy claro cuales son las comillas que dan errores, por tanto, hay que ir segmentando el código para ir probando y evaluar qué es lo que te da error. ¿Tienes algún ejemplo?

Por ejemplo hoy la Cami uso `` para escribir un texto con variable `esto es un ejemplo ${VARIABLE}` y así mismo todo el código estaba entre triple comillas ``` ``` .  Aquí la Cami estuvo agregando linea por linea hasta que se diera el error genérico. 
Incluso no tengo claro cuando usar   ``` y ```. Hay veces que al usarlo funciona, y otras veces no hay que usarlo. 
Lamentablemente, no tengo el ejemplo de ese entonces.

> Incluso sería genial agregar en la documentación las cosas que no podría sostener el ccjs. ¿Tienes una idea de cómo podría verse esto?

Por ejemplo: te recomendamos usar ``` ``` para poder aceptar los casos en que se usa .... , o bien si usar, ... es preferible escribirlo directo. 
Creo que ayudaría para que el usuario entienda que se guarda como String, por tanto, el uso de triple comillas ayuda a tomar ciertos casos.  Entiendo que el enrredo de comillas es super puntual, entonces sería genial tenerlo documentado como ayuda.


Gracias eddi!!
